compilers/concrete-compiler/compiler/lib/Dialect/FHELinalg/IR/FHELinalgOps.cpp

```diff
@@ -1687,6 +1687,494 @@ void ToUnsignedOp::getCanonicalizationPatterns(
+    std::vector<int64_t> tableContent;
+
+    TluIdexer(int64_t tableSize, bool isSigned, mlir::Value tableValue,
+              arith::ConstantOp tableOperation,
+              mlir::DenseIntElementsAttr tableContentAttr,
+              std::vector<int64_t> tableContent)
+        :
+
+          Indexer{tableSize, isSigned}, tableValue{tableValue},
+          tableOperation{tableOperation}, tableContentAttr{tableContentAttr},
+          tableContent{std::move(tableContent)} {}
+
+    ~TluIdexer() override = default;
+
+    static std::optional<std::unique_ptr<TluIdexer>>
+    create(ApplyLookupTableEintOp operation) {
+
+      auto tableValue = operation.getLut();
+
+      auto tableOperation =
+          llvm::dyn_cast_or_null<arith::ConstantOp>(tableValue.getDefiningOp());
+      if (!tableOperation) {
+        return std::nullopt;
+      }
+
+      auto tableContentAttr =
+          tableOperation.getValueAttr()
+              .dyn_cast_or_null<mlir::DenseIntElementsAttr>();
+      if (!tableContentAttr) {
+        return std::nullopt;
+      }
+
+      auto tableContent = std::vector<int64_t>();
+      for (auto value : tableContentAttr.getValues<int64_t>()) {
+        tableContent.push_back(value);
+      }
+
+      auto inputValue = operation.getT();
+      auto inputType = inputValue.getType()
+                           .cast<RankedTensorType>()
+                           .getElementType()
+                           .dyn_cast<FheIntegerInterface>();
+
+      auto tableSize = 1 << inputType.getWidth();
+      auto isSigned = inputType.isSigned();
+
+      return std::make_unique<TluIdexer>(
+          TluIdexer(tableSize, isSigned, tableValue, tableOperation,
+                    tableContentAttr, tableContent));
+    };
+
+    int64_t get(int64_t index, int64_t position) const override {
+      return tableContent[sanitizeIndex(index)];
+    }
+  };
+
+  class MappedTluIdexer : public Indexer {
+  public:
+    mlir::Value tablesValue;
+    arith::ConstantOp tablesOperation;
+    mlir::DenseIntElementsAttr tablesContentAttr;
+    std::vector<int64_t> tablesContent;
+
+    mlir::Value mapValue;
+    arith::ConstantOp mapOperation;
+    mlir::DenseIntElementsAttr mapContentAttr;
+    std::vector<int64_t> mapContent;
+
+    MappedTluIdexer(int64_t tableSize, bool isSigned, mlir::Value tablesValue,
+                    arith::ConstantOp tablesOperation,
+                    mlir::DenseIntElementsAttr tablesContentAttr,
+                    std::vector<int64_t> tablesContent, mlir::Value mapValue,
+                    arith::ConstantOp mapOperation,
+                    mlir::DenseIntElementsAttr mapContentAttr,
+                    std::vector<int64_t> mapContent)
+        :
+
+          Indexer{tableSize, isSigned}, tablesValue{tablesValue},
+          tablesOperation{tablesOperation},
+          tablesContentAttr{tablesContentAttr},
+          tablesContent{std::move(tablesContent)}, mapValue{mapValue},
```

Comment:
Storing the range from the attribute's `getValue()` method or an `ArrayRef` is probably better here, since this avoids an entire copy of an array of values whose lifetime spans beyond the lifetime of the indexer.